src/app/server-cluster/DefaultServerCluster.h

```diff
@@ -73,9 +81,26 @@ class DefaultServerCluster : public ServerClusterInterface
     /// Default implementation is a NOOP (no list items generated)
     CHIP_ERROR GeneratedCommands(const ConcreteClusterPath & path, DataModel::ListBuilder<CommandId> & builder) override;
 
+    /// Returns all global attributes that the spec defines in `7.13 Global Elements / Table 93: Global Attributes`
+    static Span<const DataModel::AttributeEntry> GlobalAttributes();
+
 protected:
+    ServerClusterContext * mContext = nullptr;
+
     void IncreaseDataVersion() { mDataVersion++; }
 
+    /// Marks that a specific attribute has changed value
+    ///
+    /// This increases cluster data version and if a cluster context is available it will
+    /// notify that the attribute has changed.
+    void NotifyAttributeChanged(AttributeId attributeId);
+
+    /// A "bulk update" notification, that notifies a wildcard attribute change.
+    ///
+    /// Increases cluster data version and if a cluster context is available, it will notify
+    /// that the cluster has changed.
+    void NotifyAllAttributesChanged();
```

Comment:
When and by whom would this ever be used?

Reply:
I thought we would need it given that we support wildcards in `Temporary_ReportAttributeChanged` ... however I see that is for full endpoint changes only. Will drop for now and figure out if we need it later.